### PR TITLE
Update README.md now that KCA works for MSA

### PR DESCRIPTION
**Pull request by jmprieur**
## Purpose
Declaring the client application as the "known client application" of the service application enables consent for the Web API to be rolled-up when the user signs-in and consents to the client. This feature is now available for MSA, and therefore the previous limitations don't apply any longer. 

This PR updates the README.md to remove the limitations and update the instructions.
It also sets the client default config file to use:
- tenant = "common" (as this now works with Personal accounts)
- todoListServiceScope = {middleTier}/.default.

It also makes it clear that the 3rd chapter is a different architecture (with the same ClientID)
* ...

## Does this introduce a breaking change?

```
[ ] Yes
[ x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
```

* Test the code

```
go through the README.md
```

### CODE REVIEW

**Comment by jmprieur** [comment]
@TiagoBrenck : this should answer one of your questions about why chapter 2 has instructions for MSA if we have a dedicated chapter 3. Now chapter 3 is a different architecture and this is well indicated (in the README.md, and through the overview picture)

@kalyankrishna1 : in case you were wondering, I was very careful to NOT rename the folders (in particular the 3rd folder which should be "web-api-now-calls-graph-same-clientID" or something like that: I don't want this PR to be poluted by renamings that will bring-in 100 file changes. We'll do the renaming part of another PR when this one is merged.

File: 2. Web API now calls Microsoft Graph/README-incremental-instructions.md [hunk 1]
```diff
@@ -123,32 +123,34 @@ These instructions only show the differences with the first part.
    - Click the **Add a permission** button and then,
    - Ensure that the **Microsoft APIs** tab is selected
    - In the *Commonly used Microsoft APIs* section, click on **Microsoft Graph**
-   - In the **Delegated permissions** section, ensure that the right permissions are checked: **User.Read**. Use the search box if necessary.
+   - In the **Delegated permissions** section, ensure that the right permissions are checked: **User.Read** and **offline_access**. Use the search box if necessary.
```
Comment (jmprieur) [inline 2. Web API now calls Microsoft Graph/README-incremental-instructions.md]: This is needed for MSA

File: 2. Web API now calls Microsoft Graph/README-incremental-instructions.md [hunk 2]
```diff
@@ -158,12 +160,9 @@ Nothing more to do more here. All was done in the first part
 
 Clean the solution, rebuild the solution, and run it
 
-### Current limitations
+### Alternative architecture
```
Comment (jmprieur) [inline 2. Web API now calls Microsoft Graph/README-incremental-instructions.md]: The limitation has gone now. the 3rd chapter proposes an interesting alternative architecture where client and service have the same client ID.

File: 2. Web API now calls Microsoft Graph/README.md [hunk 3]
```diff
@@ -163,10 +161,10 @@ If you want to register your apps manually, as a first step you'll need to:
    - Click the **Add a permission** button and then,
    - Ensure that the **Microsoft APIs** tab is selected
    - In the *Commonly used Microsoft APIs* section, click on **Microsoft Graph**
-   - In the **Delegated permissions** section, ensure that the right permissions are checked: **User.Read**. Use the search box if necessary.
+   - In the **Delegated permissions** section, ensure that the right permissions are checked: **User.Read** and **offline_access**. Use the search box if necessary.
```
Comment (jmprieur) [inline 2. Web API now calls Microsoft Graph/README.md]: Needed for MSA

File: 2. Web API now calls Microsoft Graph/README-incremental-instructions.md [hunk 4]
```diff
@@ -95,14 +96,13 @@ If you want to use this automation:
       - in the manifest, search for **"accessTokenAcceptedVersion"**, and replace **null** by **2**. This property lets Azure AD know that the Web API accepts v2.0 tokens
       - search for **signInAudience** and make sure it's set to **AzureADandPersonalMicrosoftAccount**
       - Select **Save**
-   1. In the **Authentication** page for the *TodoListService-v2* application, check the `urn:ietf:wg:oauth:2.0:oob` reply URI so that the client can propose incremental consent to the user for the Web API when needed.
    1. In tthe application registration page for the *TodoListClient-v2* application, select the **Manifest** section:
```
Comment (TiagoBrenck) [inline 2. Web API now calls Microsoft Graph/README-incremental-instructions.md]: Typo: "In tthe application"

**Comment by TiagoBrenck** [comment]
@jmprieur I tried using my other AAD accounts and had a really buggy and weird experience:

1st account showed the consent screen 2 times, then I got prompt with an error saying that an admin hasn't consented. After rebuilding the app, login in with the same account worked.

2nd account is even more buggy. The consent screen showed 7 times. After the 7th acceptance, it logged me in and I could consume the API.

Then I clicked on 'Clean Cache' and tried to login with my MSA account again, and got the error:
```
Sorry, but we’re having trouble signing you in.
 AADSTS50020: User account 'tiag0_fabri@hotmail.com' from identity provider 'live.com' does not exist in tenant 'Default Directory' and cannot access the application '84d82b8f-33fe-4e07-ba29-0eab7f393abe'(TodoListClient-v2) in that tenant. The account needs to be added as an external user in the tenant first. Sign out and sign in again with a different Azure Active Directory user account. 
```

I had to rebuild the application again to make it work with MSA. I would say that switching between MSA and AAD accounts is not working properly.

**Comment by jmprieur** [comment]
@TiagoBrenck: Thanks for this update. I believe that this could be due to bug https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1365 (on the client side)

I suggest you delete the token cache file until we release this fix
(alternatively build the MSAL.NET nuget package from the master branch)

**Pull request merged**